internal/bus/accounts.go

```diff
@@ -16,45 +16,125 @@ import (
 	"lukechampine.com/frand"
 )
 
-var errAccountsNotFound = errors.New("account doesn't exist")
+var (
+	ErrAccountNotFound = errors.New("account doesn't exist")
+)
 
-type accounts struct {
-	mu     sync.Mutex
-	byID   map[rhpv3.Account]*account
-	logger *zap.SugaredLogger
-}
+type (
+	AccountStore interface {
+		Accounts(context.Context) ([]api.Account, error)
+		SaveAccounts(context.Context, []api.Account) error
+		SetUncleanShutdown(context.Context) error
+	}
+)
 
-type account struct {
-	mu               sync.Mutex
-	locks            map[uint64]*accountLock
-	requiresSyncTime time.Time
-	api.Account
+type (
+	mgr struct {
+		s      AccountStore
+		logger *zap.SugaredLogger
 
-	rwmu sync.RWMutex
-}
+		mu   sync.Mutex
+		byID map[rhpv3.Account]*account
+	}
 
-type accountLock struct {
-	heldByID uint64
-	unlock   func()
-	timer    *time.Timer
-}
+	account struct {
+		mu               sync.Mutex
+		locks            map[uint64]*accountLock
+		requiresSyncTime time.Time
+		api.Account
+
+		rwmu sync.RWMutex
+	}
+
+	accountLock struct {
+		heldByID uint64
+		unlock   func()
+		timer    *time.Timer
+	}
+)
+
+// NewAccountManager creates a new account manager. It will load all accounts
+// from the given store and mark the shutdown as unclean. When Shutdown is
+// called it will save all accounts.
+func NewAccountManager(ctx context.Context, s AccountStore, logger *zap.Logger) (*mgr, error) {
```

Comment:
Missed it the first time but the returned type here should also be exported.
Also both open PRs got conflicts.